comictaggerlib/pagelisteditor.py

```diff
@@ -370,11 +370,16 @@ def emit_front_cover_change(self) -> None:
             self.first_front_page = self.get_first_front_cover()
             self.firstFrontCoverChanged.emit(self.first_front_page)
 
-    def set_metadata_style(self, data_style: str) -> None:
+    def set_metadata_style(self, data_styles: list[str]) -> None:
         # depending on the current data style, certain fields are disabled
-        if data_style:
-            self.metadata_style = data_style
+        if data_styles:
+            styles = [metadata_styles[style] for style in data_styles]
+            enabled_widgets = []
+            for style in styles:
+                for attr in style.supported_attributes:
+                    enabled_widgets.append(attr)
```

Comment:
Lets change enabled_widgets to a set and you can can remove the inner loop:
```suggestion
        if data_styles:
            styles = [metadata_styles[style] for style in data_styles]
            enabled_widgets = set()
            for style in styles:
                enabled_widgets.update(style.supported_attributes)
```